### DISCUSSION TITLE
Transparent Color not working for DrawText

### DISCUSSION THREAD

**Discussion by fritz-net** [SixLabors/ImageSharp.Drawing#256] [Q&A] [answered]
Hey, I tried the Watermark example which works quite fine, however if I use a color with transparency I do not see any of that.
Am I'm doing something wrong?

I also tried `color = color.WithAlpha(0.5f);` which does give the same result

```
using (var img = Image.Load("fb.jpg"))
            {
                // For production application we would recommend you create a FontCollection
                // singleton and manually install the ttf fonts yourself as using SystemFonts
                // can be expensive and you risk font existing or not existing on a deployment
                // by deployment basis.
                Font font = SystemFonts.CreateFont("Arial", 10); // for scaling water mark size is largely ignored.

                var color = Color.HotPink;
                color = new Color(new Rgba32(200, 200, 200, 50));
                using (var img2 = img.Clone(ctx => ctx.ApplyScalingWaterMark(font, "A short piece of text", color, 5, false)))
                {
                    img2.Save("output/simple.png");
                }


                using (var img2 = img.Clone(ctx => ctx.ApplyScalingWaterMark(font, LongText, color, 5, true)))
                {
                    img2.Save("output/wrapped.png");
                }

                // The original `img` object has not been altered at all.
            }
```

**Comment by JimBobSquarePants** [comment, marked as answer]
Your jpeg is 24bit RGB. Using the non-generic variant `Image` will load that image as `Rgb24`. For alpha support you need to specify `Rgba32` using the non-generic `Image<TPixel>`

**Comment by fritz-net** [comment]
I thought that the alpha channel of the image is irrelevant since the color is applied anyway and there is no alpha channel on the image itself/the result is alpha 100% anyway.

Thanks for your response, I fixed it using `using (var img = Image.Load<Rgba32>("fb.jpg"))`. Using Image<Rgba32> does not work (and seems to be the default anyway)

**Reply by JimBobSquarePants** [reply]
There is no default. 

Jpegs contain no alpha channel so the non-generic `Image` for that format will represent either `Rgb24` or `L8` depending on the encoded color type. Other image formats will return `Rgba32`, `Rgba64`, `L16` `La16` or others depending on the encoded color type supported by that encoded format. These are deliberate decisions made by our decoders in order to be as efficiant as possible. 

Using the generic image `Image<T>` tells the decoder to use your format instead. When drawing with a color that contains an alpha channel you need to ensure that the surface you are drawing upon supports that channel.
